xclim/indices/_threshold.py

```diff
@@ -1928,6 +1929,58 @@ def heat_wave_index(
     return to_agg_units(out, tasmax, "count")
 
 
+@declare_units(tasmax="[temperature]", thresh="[temperature]")
+def heat_wave_magnitude(
```

Comment:
About the name. If I'm thinking about:
`heat_spell_max_length`
`heat_spell_total_length`

I think `heat_wave_magnitude` is more similar to `heat_spell_max_length`. Could there be a use for an indicator that computes the cumulative heat across the heat wave during a year, similar to what `heat_spell_total_length`?  If so, should we adopt a notation like:
`heat_wave_magnitude -> heat_wave_max_magnitude`
and potentially a new indicator in the future:
`heat_wave_total_magntitude` ?

Reply:
In fact, what I just described would compute the cumulative heat, and not the exceedance over a threshold.  Inspired by `degree_days_exceedance_date`, we could have:

`heat_wave_magnitude -> heat_wave_max_exceedance`

Does this sound clear?

Reply:
Sounds clear, yes. I was mainly calling it magnitude because that's how it's described in the literature if I am not mistaken (e.g. https://agupubs.onlinelibrary.wiley.com/doi/10.1002/2014JD022098). 

Distinguishing between and allowing for both, max and total, is a good point, since there are multiple different Heat Wave definitions (and perhaps Magnitude indices definitions) out there and I can imagine that the total quantity might be interesting to analyze as well.

Reply:
In the paper you shared, I don't see any threshold exceedance mentioned?  It seems that the definition that is often used   elsewhere is the Heat Wave Magntitude Index which is scaled between 0 and 1 (but still, it's tasmax and not tasmax -thresh that is transformed). This indicator seems to be the most standard.

Here : https://agupubs.onlinelibrary.wiley.com/doi/epdf/10.1029/2020GL087097 the heatwave threshold exceedance is referred to as heat severity (there is also a spatial aspect to this)

Here: https://agupubs.onlinelibrary.wiley.com/doi/pdfdirect/10.1029/2022EA002567 the magnitude is defined in a similar way is the Tmax anomalies, but those anomalies are normalized before-hand. 

Similarly here: https://journals.ametsoc.org/view/journals/apme/56/9/jamc-d-16-0393.1.xml?tab_body=pdf, the magntitude refers to the "average departure from the heat wave day threshold (90th percentile as defined above) among grid points experiencing heat wave conditions."

All in all, it seems that "magnitude" is indeed often used to refer to the exceedance relative to a reference value. I think we can go with this since it seems standard. 

It doesn't seem to be the case that there is a well established definition for `heat_wave_magnitude`, but `heat_wave_magnitude_index` is more standard. I think it's fine that we use `heat_wave_max_magnitude`.

Reply:
Just to say it clearly: 

I suggest changing the name of this current indicator to `heat_wave_max_magnitude`, to mimick `heat_wave_max_length`.  Magnitude is apparently often used as a synonym for "threshold exceedance" in the case of heat waves, so I think it's ok to keep this. @aulemahal @Zeitsperre if you have objections please let me know.

In the future,  `heat_wave_total_magnitude` could be considered in the future.

Reply:
I just realized it,  but "heat_wave" is used elsewhere in xclim for the double-thresholded indicator (tasmax & tasmin) while the tasmax-only threshold is called "hot spell". I fear that the current name of the new indicator would bring confusion and I would suggest renaming to  `hot_spell_[...]`.

Sorry to come with this important change so late in the process!

As for the suffix, I agree with adding a "max" to leave the door open for a "total". 

"intensity" is used in 3 other indicators of xclim, all relating to the average value of precipitation.  I thus agree with using `magnitude` here, even more considering that it's already being used in the literature.

And @coxipi, I have yet to see a meaningful use of a sum of non-anomalies temperatures!

Reply:
It makes sense for `hot_spell`, it's unfortunate to have a name that's not closer to litterature but it's the price to pay to have coherent naming within xclim. 

> meaningful use of a sum of non-anomalies temperatures

That was more to reflect on the fact that in common language, "magntitude" makes me think of the temperature directly, not the anomalies. But I'm thinking more about "heat magnitude". Maybe it makes more sense like this, if there is no "heat wave", then the "heat wave magnitude" is zero.  

And yes, I guess summing temperatures in a given heatwave doesn't make much sense, the interesting information would be: what was the mean temperature / max temperature in a given heatwave. Anyways